pkg/common/file_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/cri-tools/pkg/common"
+)
+
+var _ = DescribeTable("ReadConfig",
+	func(content string, expectedConfig *common.Config, shouldFail bool) {
+		f, err := os.CreateTemp("", "crictl-read-config-")
+		defer os.RemoveAll(f.Name())
+		Expect(err).NotTo(HaveOccurred())
+
+		_, err = f.WriteString(content)
+		Expect(err).NotTo(HaveOccurred())
+
+		readConfig, err := common.ReadConfig(f.Name())
+		if shouldFail {
+			Expect(err).To(HaveOccurred())
+			return
+		} else {
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		Expect(readConfig.RuntimeEndpoint).To(Equal(expectedConfig.RuntimeEndpoint))
+		Expect(readConfig.ImageEndpoint).To(Equal(expectedConfig.ImageEndpoint))
+		Expect(readConfig.Timeout).To(Equal(expectedConfig.Timeout))
+		Expect(readConfig.Debug).To(Equal(expectedConfig.Debug))
+		Expect(readConfig.PullImageOnCreate).To(Equal(expectedConfig.PullImageOnCreate))
+		Expect(readConfig.DisablePullOnRun).To(Equal(expectedConfig.DisablePullOnRun))
+	},
+
+	Entry("should succeed with config", `
+runtime-endpoint: "foo"
+image-endpoint: "bar"
+timeout: 10
+debug: true
+pull-image-on-create: true
+disable-pull-on-run: true
+`, &common.Config{
+		RuntimeEndpoint:   "foo",
+		ImageEndpoint:     "bar",
+		Timeout:           10,
+		Debug:             true,
+		PullImageOnCreate: true,
+		DisablePullOnRun:  true,
+	}, false),
+
+	Entry("should fail with invalid config option", `runtime-endpoint-wrong: "foo"`, nil, true),
```

Comment:
Would be great to have tests:

1. with the comments in the file
2. with empty lines between entries
3. with duplicated entries
4. empty file

All of them I believe are considered to be valid configs and a not that unusual to have

Reply:
Sure, adding them in https://github.com/kubernetes-sigs/cri-tools/pull/1648